torch/csrc/distributed/rpc/functions.cpp

```diff
@@ -50,6 +61,50 @@ Message processRequestBlocking(Message&& request) {
       }
       break;
     }
+    case MessageType::REMOTE_CALL: {
+      ScriptRemoteCall src = ScriptRemoteCall::fromMessage(request);
+
+      auto rrefId = RRefId::fromIValue(src.retRRefId());
+      auto forkId = ForkId::fromIValue(src.retForkId());
+      TORCH_CHECK(rrefId != forkId, "Does not support remote call to self.");
+
+      auto& ctx = RRefContext::getInstance();
+      auto ownerRRef = ctx->getOrCreateOwnerRRef<IValue>(rrefId);
+
+      // TODO: make this asynchronous
+      // src is only alive within this block, use reference to avoid copy
+      auto& stack = src.stackRef();
+      src.op()->getOperation()(stack);
+      AT_ASSERT(
+          stack.size() == 1,
+          "Return value of a builtin operator or a "
+          "TorchScript function should be a single IValue, got a vector of "
+          "size ",
+          stack.size());
+
+      ownerRRef->setValue(std::move(stack.front()));
+      return Message();
+    }
+    case MessageType::RREF_FETCH_CALL: {
+      ScriptRRefFetchCall srf = ScriptRRefFetchCall::fromMessage(request);
+      // TODO: make this asynchronous
+      std::shared_ptr<OwnerRRef<IValue>> rref =
+          RRefContext::getInstance()->getOrCreateOwnerRRef<IValue>(
```

Comment:
This is not a bug if we don't enforce `OwnerRRef` be created before any `UserRRef`, because the caller could have shared the `UserRRef` with other users, and those users can communicate with the owner before the caller's message arrives at the owner. 

Let's imagine a model parallel training case. Say there is a trainer that drives the training, and the model locates on 10 machines. One implementation could be the trainer make a remote call to the machine that hosts the first piece of the model, and immediately gets the RRef of the return value, then forward the RRef to the second piece of the model, and so on. If we enforce creating `OwnerRRef` before `UserRRef`, then we are forcing 10 more round-trips delay in one iteration. If we create `OwnerRRef` after the operator execution is done, we are then forcing the model to be run in a serialized manner, while the model could be two tiers of 5 parallel layers. 

It is true that the exception of those operation execution hasn't been captured properly, and they should. But I feel we don't have to create `OwnerRRef` after the operation finishes, because if there are failures, the one who retrieves the value from the `RRef` will see that error. I will add this. 

Exclude the exception handling part, then the question is do we want to trade performance for code simplicity?